File: Data Structures & Algorithms/merge-intervals/submission-0.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if len(intervals) == 1:
            return intervals
        intervals.sort()
        res = []
        curr = intervals[0]
        for interval in intervals:
            if curr[0] > interval[1]:
                res.append(interval)
            elif curr[1] < interval[0]:
                res.append(curr)
                curr = interval
            else:
                curr[0] = min(curr[0], interval[0])
                curr[1] = max(curr[1], interval[1])
        
        res.append(curr)

        return res

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
#include <vector>
#include <iostream>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l, r, curr, max;
        l = curr = max = 0;
        r = heights.size() - 1;

        while(l < r){
            curr = min(heights[l], heights[r]) * (r-l);
            if(curr > max) max = curr;

            if(heights[l] <= heights[r]) l++;
            else r--;
        }
        return max;
    }
};
